components/higher-order/with-lazy-dependencies/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+const alreadyLoaded = {};
+
+function loadScript( url ) {
+	if ( alreadyLoaded[ url ] ) {
```

Comment:
We perform `url` based checks but pass a list of registered assets. What if there are two concurrent component being loaded at the same time where the same asset is being requested? Example:

```js
withLazyDependencies( { scripts: [ 'foo', 'bar' ] } ( ComponentA );
withLazyDependencies( { scripts: [ 'foo', 'dummy' ] } ( ComponentB );
```
Should we care? I guess it depends on what we load, if scripts have side-effects then we probably should avoid loading it twice.

Reply:
Do we need to solve it now? Not necessarily. I'm sharing so we were ready to take that into account when more than one component uses this HOC.

Reply:
Good catch! We should change this to check handles rather than URLs.